dali/test/python/test_pipeline_decorator.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline, pipeline_def
+from nose.tools import nottest, raises
+import nvidia.dali.fn as fn
+from test_utils import get_dali_extra_path, compare_pipelines
+import os
+
+data_root = get_dali_extra_path()
+images_dir = os.path.join(data_root, 'db', 'single', 'jpeg')
+
+N_ITER = 7
+
+max_batch_size = 16
+num_threads = 4
+device_id = 0
+
+
+def reference_pipeline(flip_vertical, flip_horizontal, ref_batch_size=max_batch_size):
+    pipeline = Pipeline(ref_batch_size, num_threads, device_id)
+    with pipeline:
+        data, _ = fn.file_reader(file_root=images_dir)
+        img = fn.image_decoder(data, device="mixed")
+        flipped = fn.flip(img, horizontal=flip_horizontal, vertical=flip_vertical)
+        pipeline.set_outputs(flipped, img)
+    return pipeline
+
+
+@nottest
+@pipeline_def(batch_size=max_batch_size, num_threads=num_threads, device_id=device_id)
+def pipeline_static(flip_vertical, flip_horizontal):
```

Comment:
We're missing the tests that were discussed:
* 0 returns from function (or `None`) 
* 1 return from function
* list instead of tuple?
set_outputs() probably handles some constants, but maybe it's good to test an `int`, `string`, lambda or some other weirdness.

Reply:
I wonder what happens when `set_outputs` has 0 arguments...

Reply:
We agreed, that the decorator should pass everything to `set_outputs` and `set_outputs` should validate correctness. This way, the only question is about having
```python
return
return None
```
and how to deal with this. Added specific `if` clause to handle this case.